torch/jit/python_frontend.py

```diff
@@ -0,0 +1,285 @@
+import torch
+import sys
+import ast
+import inspect
+from functools import partial
+from collections import namedtuple
+from torch._C._jit_tree_views import *
+
+# TODO: improve error reporting (show source)
+#       once this is done, convert all asserts into checks with nicer error messages
+
+PY2 = sys.version_info[0] == 2
+_reserved_prefix = '__jit'
+
+
+def get_jit_ast(fn):
+    source = dedent(inspect.getsource(fn))
+    py_ast = ast.parse(source)
+    if len(py_ast.body) != 1 or not isinstance(py_ast.body[0], ast.FunctionDef):
+        raise RuntimeError("expected a single top-level function")
+    return build_def(SourceRangeFactory(source), py_ast.body[0])
+
+
+def dedent(source):
+    lines = source.split('\n')
+    indent_depth = 0
+    while lines[0][indent_depth] == ' ':
+        indent_depth += 1
+    return '\n'.join(l[indent_depth:] for l in lines)
+
+
+class Builder(object):
+    def __call__(self, ctx, node):
+        try:
+            method = getattr(self, 'build_' + node.__class__.__name__)
+        except AttributeError:
+            raise RuntimeError(node.__class__.__name__ + " isn't a supported Python feature")
+        return method(ctx, node)
+
+
+class CountReturns(ast.NodeVisitor):
+    def __init__(self):
+        self.num_returns = 0
+
+    def visit_Return(self, ret):
+        self.num_returns += 1
+
+    @staticmethod
+    def get_count(py_def):
+        counter = CountReturns()
+        counter.visit(py_def)
+        return counter.num_returns
+
+
+_ret_err_msg = ("JIT-ed functions can only have a single return, "
+                "and it has to be the last statement in the body")
+
+
+def build_def(ctx, py_def):
+    assert len(py_def.decorator_list) == 0
+    returns = []
+    ret_body = []
+    body = py_def.body
+    num_returns = CountReturns.get_count(py_def)
+    if num_returns == 1:
```

Comment:
Let's change the AST to have a Return statement and change the script frontend as well. We can check in compile that the Return statement only appears as the last statement of the function. This will be more consistent with what we want the behavior to be in the future.